xray/services/report.go

```diff
@@ -122,12 +179,27 @@ func NewReportService(client *jfroghttpclient.JfrogHttpClient) *ReportService {
 }
 
 // Vulnerabilities requests a new Xray scan for vulnerabilities
-func (rs *ReportService) Vulnerabilities(req ReportRequestParams) (*ReportResponse, error) {
+func (rs *ReportService) Vulnerabilities(req VulnerabilitiesReportRequestParams) (*ReportResponse, error) {
+	return rs.requestReport(req, Vulnerabilities)
+}
+
+// Licenses requests a new Xray scan for licenses
+func (rs *ReportService) Licenses(req LicensesReportRequestParams) (*ReportResponse, error) {
+	return rs.requestReport(req, Licenses)
+}
+
+// Violations requests a new Xray scan for violations
+func (rs *ReportService) Violations(req ViolationsReportRequestParams) (*ReportResponse, error) {
```

Comment:
Looks like the new APIs are not covered with tests. We are unable to add new code without tests.
Let's add some tests "reportLicenses" and "reportViolaions" in xrayreport_test.go to cover this new logic.

Reply:
@yahavi I have modified existing test testfile and added new test for license xray report api.